src/runtime/compat.rs

```diff
@@ -17,27 +17,61 @@ pub(super) struct Background {
     thread: Option<thread::JoinHandle<()>>,
 }
 
-#[derive(Debug)]
-pub(super) struct Now<N>(N);
+#[derive(Clone, Debug)]
+pub(super) struct CompatSpawner<S> {
+    pub(super) inner: S,
+    pub(super) idle: idle::Idle,
+}
 
-#[derive(Debug)]
-struct CompatPark<P>(P);
+struct CompatGuards {
+    _executor: executor_01::DefaultGuard,
+    _timer: timer_02::DefaultGuard,
+    _reactor: reactor_01::DefaultGuard,
+}
 
-impl Background {
-    pub(super) fn spawn(clock: &tokio_02::timer::clock::Clock) -> io::Result<Self> {
-        let clock = clock_02::Clock::new_with_now(Now(clock.clone()));
+thread_local! {
+    static COMPAT_GUARDS: RefCell<Option<CompatGuards>> = RefCell::new(None);
+}
+
+pub(super) fn set_guards(
+    executor: impl executor_01::Executor + 'static,
+    timer: &timer_02::Handle,
+    reactor: &reactor_01::Handle,
+) {
+    let guards = CompatGuards {
+        _executor: executor_01::set_default(executor),
+        _timer: timer_02::set_default(timer),
+        _reactor: reactor_01::set_default(reactor),
+    };
+
+    COMPAT_GUARDS.with(move |current| {
+        let prev = current.borrow_mut().replace(guards);
+        assert!(
```

Comment:
do we want to debug_assert?

Reply:
tokio will panic if something is entered twice even in release mode...

Reply:
will it panic even if this assert isn't here just thinking it may be expensive to call this in a hot path like this no?